src/components/modals/CreateForm.tsx

```
@@ -0,0 +1,139 @@
+import { useDispatch } from "react-redux"
+import { useNavigate } from "react-router-dom"
+
+interface CreateFormModalProps {
+  isOpen: boolean
+  onClose: () => void
+  formType: FormType
+}
+
+const CreateFormSchema = z.object({
+  title: z.string().min(1, "Title is required"),
+  description: z.string().optional(),
+})
+
+type CreateFormInput = z.infer<typeof CreateFormSchema>
+
+export default function CreateForm({
+  isOpen,
+  onClose,
+  formType,
+}: CreateFormModalProps) {
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm<CreateFormInput>({
+    resolver: zodResolver(CreateFormSchema),
+  })
+
+  const dispatch = useDispatch()
+  const navigate = useNavigate()
+  const [cookies] = useCookies([Cookie.jwt])
+  const [createForm, { isLoading }] = useCreateFormMutation()
+
+  const onSubmit = async (data: CreateFormInput) => {
+    const requestBody: {
+      name: string;
+      type: FormType;
+      description?: string;
+      startDate?: string;
+      endDate?: string;
+      stages?: { name: string }[];
+    } = {
+      name: data.title,
+      type: formType,
+    }
+
+    if (data.description) {
+      requestBody.description = data.description
+    }
+
+    if (formType === FormType.Application) {
+      const now = new Date()
+      requestBody.startDate = new Date(
+        now.getTime() + 24 * 60 * 60 * 1000,
+      ).toISOString()
+      requestBody.endDate = new Date(
+        now.getTime() + 48 * 60 * 60 * 1000,
+      ).toISOString()
+      requestBody.stages = [{ name: "Application" }]
+    }
+
+    try {
+      const result = await createForm({
+        jwt: cookies.jwt,
+        body: requestBody,
+      }).unwrap()
+      navigate(`/forms/${result._id}?edit=true`)
+    } catch (error) {
+      const { message } = getErrorInfo(error)
+      handleShowAlert(dispatch, {
+        type: AlertType.Error,
+        message,
+      })
+    } finally {
+      reset()

```

`reset` should only happen if the submission is successful. I think it should be placed in try block. This is because all errors will be lost when submission fails.